pkg/operator/applier/kubectl.go

```diff
@@ -132,6 +136,138 @@ commonAnnotations:
 	return stdout, stderr, errors.Wrap(err, "failed to run kubectl apply")
 }
 
+// ApplyCreateOrPatch attempts to run a `kubectl apply` on the yaml document. If it fails
+func (c *Kubectl) ApplyCreateOrPatch(targetNamespace string, slug string, yamlDoc []byte, dryRun bool, wait bool, annotateSlug bool) ([]byte, []byte, error) {
+
+	stdout, stderr, err := c.Apply(targetNamespace, slug, yamlDoc, dryRun, wait, annotateSlug)
+	if !strings.Contains(string(stderr), "metadata.annotations: Too long") || err == nil {
+		return stdout, stderr, errors.Wrap(err, "failed attempted kubectl apply")
+	}
+	log.Println("Failed to apply document: metadata too long. Trying to 'create' or 'patch' instead")
+
+	docs, err := splitter.SplitYAML(yamlDoc)
```

Comment:
kubectl create accepts multi-yaml docs, why do we need to split them here?

Reply:
the current kubectl apply workflow doesn't split them, so i don't think this should

Reply:
Yes, but patch does not. I didn't want to get into a scenario where create only partially fails (like maybe someone added new manifest to an existing doc), then we have to figure out which documents we need to patch anyway. I also opted to inline them here to minimize the io of writing out this files and the temp dir.

Reply:
ah okay